t/t5516-fetch-push.sh

```diff
@@ -905,7 +908,8 @@ test_expect_success 'push with branches' '
 
 test_expect_success 'push with branches containing #' '
 	mk_empty testrepo &&
-	echo "testrepo#branch3" > .git/branches/branch2 &&
```

Comment:
Hi dscho, I don't see why this appears as deleted here, but not in the cases above. Is there a spelling or white space change I'm not seeing....

seen it: its the space after the `>` redirection. I'm also presuming that, long term, these (simple) fixes will eventually also go upstream.

Reply:
Oh crap, I meant to add "While at it, we fix white-space issues: in Git's shell script, we do not want a space between the redirector (`>`) and the file to which to redirect to.".

Reply:
Extracted into its own commit.